sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_response.py

```diff
@@ -0,0 +1,22 @@
+# The MIT License (MIT)
+# Copyright (c) 2024 Microsoft Corporation
+
+from azure.core.utils import CaseInsensitiveDict
+
+
+class CosmosResponse(dict):
+    def __init__(self, original_dict, response_headers):
+        super().__init__(original_dict)
+        self._response_headers = response_headers
+
+    def response_headers(self) -> CaseInsensitiveDict:
```

Comment:
I would make this a property rather than a method:
```suggestion
    @property
    def response_headers(self) -> CaseInsensitiveDict:
```